api/base/serializers.py

```diff
@@ -857,17 +857,20 @@ def to_representation(self, value):
         self_url = url['self']
         self_meta = self.get_meta_information(self.self_meta, value)
         relationship = format_relationship_links(related_url, self_url, related_meta, self_meta)
-        if related_url and (len(related_path.split('/')) & 1) == 1:
+        if related_url:
```

Comment:
Here's where this was added: https://github.com/CenterForOpenScience/osf.io/commit/cc6ba73d14cb24f1c3d34694a5493165a2c9f9b2

@aaxelb, is this important to keep?

Reply:
As far as i can tell, this is fine. My best guess is that counting path segments was an earlier attempt at detecting whether the relation points to a detail view, before the `issubclass(related_class, RetrieveModelMixin)` solution was found.